2.0.0

sixtsense opened the pull request.

Commit:
2.0.0 | Update project dependencies

 - Set project version to 2.0.0
 - Bump gradle wrapper to 8.13
 - Bump gradle to 8.9.1
 - Set `sourceCompatibility` and `targetCompatibility` to Java 11
 - Bump NDK version to `29.0.13113456`
 - Set `compileSdkVersion`, `targetSdkVersion` to 35
 - Set `minSdkVersion` to 24

Commit:
2.0.0 | Update native files

 - Update `libmodpdfium.so` libs with 16kb versions
 - Update supporting public `*.h` files
 - Fix problem in `mainJNILib.cpp`

Commit:
2.0.0 | Add guide for building pdfium native lib

georgi-neykov-hub commented on BUILD_NATIVE.md:
```diff
@@ -0,0 +1,76 @@
+
+## Instructions
+
+1. **Copy `build_pdfium.sh` to a Linux-based virtual machine:**
```
You can possibly leave out the virtual machine part, there will be people trying to do a build on a bare metal Linux host or in a container.

georgi-neykov-hub commented on BUILD_NATIVE.md:
````diff
@@ -0,0 +1,76 @@
+      ```
+
+3. **Run the script:**
+    - Execute the script and wait for it to finish. This will clone the latest stable branch of the PDFium repository into the specified directory along with all required dependencies and third-party libraries.
````
If the target version/branch/tag is not specified, the version of pdfium being pulled from upstream may be different across script executions. Think about allowing to pass a tag name as a script parameter to the script. 
PDFium is using the same build tools as the Chromium project, here's a [guide](https://www.chromium.org/developers/how-tos/get-the-code/working-with-release-branches/#syncing-and-building-a-release-tag) on how to checkout a specific version from there.

georgi-neykov-hub commented on BUILD_NATIVE.md:
````diff
@@ -0,0 +1,76 @@
+      ```gn
+      cflags = [
+          "-fvisibility=default",
+          "-Oz",  # This reduces the size of the final binary
````
This flag is only appropiate if building a "release" variant as the binaries will not be debug-friendly. Add a note about this, and possibly let users pick the code optimization levels themselves. 
Here's a reference on the optimization flags' meanings for clang/llvm:
https://clang.llvm.org/docs/CommandGuide/clang.html#code-generation-options

georgi-neykov-hub commented on BUILD_NATIVE.md:
```diff
@@ -0,0 +1,76 @@
+      ]
+      ldflags = [
+          "-Wl,-export-dynamic",      # Export dynamic symbols
+          "-Wl,-z,max-page-size=16384",  # Set max page size
```
Leave a note about this being required by Android 15+ build to allow supporting 16kB page cache sizes.
A link to the announcement form Google with links to instructions, etc:
https://android-developers.googleblog.com/2024/08/adding-16-kb-page-size-to-android.html

georgi-neykov-hub commented on src/main/jni/include/fpdf_attachment.h:
```diff
@@ -0,0 +1,179 @@
+// Copyright 2017 The PDFium Authors
```
It would be good if we at least add a README.md for the `include` folder to describe the origin of the header files here, like at least specify a tag/revision of the PDFium repository the headers have been copied.

Commit:
2.0.0 | Fix merge request comments

 - Update `build.pdfium.sh` to close a chose `githash` for the pdfium build
 - Add additional information to the `BUILD_NATIVE.md`

Commit:
2.0.0 | Add add script ot build pdfium as static lib

 - Add `build_static_lib.sh` script dedicated to build pdfium as a static lib
 - Update the `BUILD_NATIVE.md` to take in to account the two different build scripts
